### ISSUE TITLE
New verb for cnc-ddraw

### ISSUE THREAD

**Issue by Gcenx** [Winetricks/winetricks#1156]
Could we get a new verb for install [cnc-ddraw](https://github.com/CnCNet/cnc-ddraw) as the last recent version that it worked good in was wine 3.4 after that it has remained broken again.

[Wine bug](https://bugs.winehq.org/show_bug.cgi?id=33211)

Affected games are;
Command & Conquer Gold
Command & Conquer: Red Alert
Command & Conquer: Tiberian Sun
Command & Conquer: Red Alert 2
Carmageddon
Carmageddon 2
Warcraft 2
StarCraft
Diablo 2
Age of Empires
Age of Empires II
Fallout
Theme Hospital
Populous: The Beginning
Outlaws
Dungeon Keeper
Dark Reign: The Future of War
Star Wars: Galactic Battlegrounds
Atomic Bomberman
Dune 2000
Oddworld: Abe's Oddysee
Commandos
Red Baron 3D
F-16 Multirole Fighter
F-22 Raptor
Nox

With cnc-ddraw installed and set as native I'm able to play Command & Conquer: Red Alert 2 Yuris Revenge again at full speed using any wine version.

**Comment by austin987** [comment]
Do any of these have a free demo?

It's interesting that the wine bug doesn't mention this..

**Comment by Gcenx** [comment]
I’m not sure if any of games have a demo anymore but I do know of the multiplayer only clients that were for download from [Xwis licenses Multiplayer Clients](https://xwis.net/forums/topic/177134-downloads/) but those links are now dead so [Xwis wayback](https://web.archive.org/web/*/https://xwis.net/downloads/*)
Red Alert 2 & Yuri’s Revenge, those are licensed by EA and where a lot of people still play Red Alert 2 to this day. Yuris Revenget players moved into CNCNet clients 

[Tiberian Sun is freeware](https://en.m.wikipedia.org/wiki/Command_%26_Conquer:_Tiberian_Sun) [download](https://download.cnet.com/Command-Conquer-Tiberian-Sun/3000-7483_4-10037653.html)

The bug has been around before cnc-ddraw was a project so I guess nobody bothered since it seemed more of a won’t fix bug at this point

**Comment by austin987** [comment]
Thanks @Gcenx; the download link in particular is helpful. I'll look in the coming days.

Note that this isn't a WONTFIX upstream; if it was, someone would've marked it as such. There are a lot of bugs and limited manpower, particularly in d3d. But if winetricks can provide an easy workaround, I'm happy to add one.

**Comment by austin987** [comment]
@Gcenx could you try https://github.com/austin987/winetricks/tree/cnc_ddraw?

This doesn't work for me; unfortunately, I get no mouse in the Tiberian Sun demo.

**Comment by Gcenx** [comment]
@austin987 I got that working for RA2 and YR on my system.
cnc-ddraw only functions if the exact exactable name is used. If I launch YR with gamemd.exe then it will function correctly if I launched using another exactable name then it won't function.

So I guess the TS Demo you tried is not listed in the ddraw.ini file?

Yeah I know it's not a WONTFIX according to the bug but it _seems_ like it as its was broken a good while ago, but now having this added will get at least some games functioning again.
